lib/smart_answer_flows/calculate-your-holiday-entitlement.rb

```diff
@@ -210,128 +233,61 @@ def define
 
       # Q26 - Q32
       value_question :shift_worker_hours_per_shift?, parse: Float do
-        calculate :hours_per_shift do |response|
-          hours_per_shift = response
-          raise InvalidResponse, :error_no_hours_worked if hours_per_shift <= 0
-          raise InvalidResponse, :error_over_24_hours_worked if hours_per_shift > 24
+        on_response do |response|
+          calculator.hours_per_shift = response
+        end
+
+        validate :error_no_hours_worked do
+          calculator.hours_per_shift.positive?
+        end
 
-          hours_per_shift
+        validate :error_over_24_hours_worked do
+          calculator.hours_per_shift.between?(1, 24)
```

Comment:
As the lower limit is already covered in the `:error_no_hours_worked` validate block, I think this should just cover the condition where `hours_per_shift` is greater that 24 as the title suggests. So
```ruby
calculator.hours_per_shift <= 24
```

Reply:
Updated as per suggestion - thanks 👍